engine/src/cmd/unit_generic.h

```diff
@@ -613,16 +603,14 @@ class Unit : public Armed, public Audible, public Drawable, public Damageable, p
         return ToLocalCoordinates((un->Position() - Position()).Cast());
     }
 
-//how visible the ship is from 0 to 1
+// how visible the ship is from 0 to 1
+// Need this function to expose it to python
     float CloakVisible() const {
-        if (cloaking < 0) {
-            return 1;
-        }
-        return ((float) cloaking) / 2147483647;
+        cloak.Visible();
```

Comment:
linter raising an issue here since it does need to return a value or be refactored to be void.
Perhaps this should be `return clock.Visible();`?

Reply:
Yep.

Reply:
> Would be good to have to some instructions for testing this.
1. Visual - switch to view 6. Hit c and watch the ship slowly cloak itself and decloak itself.
2. Pick a fight and then cloak yourself and check the enemy doesn't shoot you. He may accidentally ram you...
3. Buy a cloaking device. Sell a cloaking device. The whole upgrade issue has not been tested properly.

Reply:
@royfalk where can I buy a cloaking device? I have yet to see one.

Reply:
@BenjamenMeyer Try buying a cloaking device at an Aera planet or base, maybe? I seem to recall someone saying at some point that Aera bases and planets tend to have cloaking devices, even if others don't.

Reply:
@stephengtuggy thanks for the suggestion...I'll have to try that out